chpl/chpl-api/src/main/java/gov/healthit/chpl/web/controller/FilesController.java

```diff
@@ -0,0 +1,96 @@
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.multipart.MaxUploadSizeExceededException;
+import org.springframework.web.multipart.MultipartFile;
+
+import gov.healthit.chpl.dto.FilesDTO;
+import gov.healthit.chpl.exception.EntityCreationException;
+import gov.healthit.chpl.exception.EntityRetrievalException;
+import gov.healthit.chpl.exception.ValidationException;
+import gov.healthit.chpl.manager.FilesManager;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+
+@Api(value = "files")
+@RestController
+@RequestMapping("/files")
+public class FilesController {
+    private static final String APPLICATION_MS_EXCEL =
+            "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet";
+
+    private FilesManager filesManager;
+
+    @Autowired
+    public FilesController(final FilesManager filesManager) {
+        this.filesManager = filesManager;
+    }
+
+    @ApiOperation(value = "Upload an API Documenation file",
+            notes = "Uploads a new current API Documentation file.")
+    @RequestMapping(value = "/api_documentation",
+            method = RequestMethod.POST,
+            produces = "application/json; charset=utf-8")
+    public synchronized ResponseEntity<FilesDTO> uploadApiDocumentation(
+            final @RequestParam("file") MultipartFile file,
+            final @RequestParam("file_update_date") Long date)
+            throws EntityRetrievalException, EntityCreationException, ValidationException,
+            MaxUploadSizeExceededException, IOException {
+
+        if (file.isEmpty()) {
+            throw new ValidationException("You cannot upload an empty file!");
+        }
+
+        FilesDTO newFileDTO = new FilesDTO();
+        newFileDTO.setAssociatedDate(new Date(date));
+        newFileDTO.setContentType(file.getContentType());
+        newFileDTO.setFileName(file.getOriginalFilename());
+        newFileDTO.setFileData(file.getBytes());
+
+        FilesDTO fileDTO = filesManager.addApiDocumentationFile(newFileDTO);
+
+        return new ResponseEntity<FilesDTO>(fileDTO, HttpStatus.OK);
+    }
+
+    @ApiOperation(value = "Retrieve an API Documenation file",
+            notes = "Retrieves the current API Documenation file.")
+    @RequestMapping(value = "/api_documentation", method = RequestMethod.GET, produces = APPLICATION_MS_EXCEL)
+    public synchronized ResponseEntity<byte[]> getApiDocumentationFile() throws EntityRetrievalException {
+        FilesDTO fileDTO = filesManager.getApiDocumentation();
+
+        String filename = "APIDocData-" + getDateAsYYYYMMDD(fileDTO.getAssociatedDate()) + ".xlsx";
```

Comment:
How sure are you that the uploaded file will support being an `.xlsx` type of file on download?